File: src/platform/process-other.ts
import net from "node:net";
import type { ProcessDiscoverer, ProcessInfo } from "./interface.js";

export class OtherProcessDiscoverer implements ProcessDiscoverer {
  async findProcesses(): Promise<ProcessInfo[]> {
    // Not supported on this platform
    return [];
  }

  async findProcessByPort(_port: number): Promise<ProcessInfo | null> {
    // Not supported on this platform
    return null;
  }

  async findProcessByName(_name: string): Promise<ProcessInfo[]> {
    // Not supported on this platform
    return [];
  }

  async isPortInUse(port: number): Promise<boolean> {
    // Basic check using net module
    return new Promise((resolve) => {
      const server = net.createServer();

      server.once("error", () => resolve(true));
      server.once("listening", () => {
        server.close();
        resolve(false);
      });

      server.listen(port);
    });
  }
}

File: internal/platform/process_other.go
//go:build !linux && !darwin && !windows

package platform

import (
	"fmt"
	"runtime"
)

// NewProcessDiscoverer creates a process discoverer for unsupported platforms
func NewProcessDiscoverer() ProcessDiscoverer {
	return &basicProcessDiscoverer{}
}

// basicProcessDiscoverer implements ProcessDiscoverer for unsupported platforms
type basicProcessDiscoverer struct{}

// FindProcesses returns an empty list on unsupported platforms
func (b *basicProcessDiscoverer) FindProcesses() ([]ProcessInfo, error) {
	return nil, fmt.Errorf("process discovery not supported on %s", runtime.GOOS)
}

// FindProcessByPort returns nil on unsupported platforms
func (b *basicProcessDiscoverer) FindProcessByPort(port int) (*ProcessInfo, error) {
	return nil, fmt.Errorf("process discovery not supported on %s", runtime.GOOS)
}

// FindProcessByName returns nil on unsupported platforms
func (b *basicProcessDiscoverer) FindProcessByName(name string) ([]ProcessInfo, error) {
	return nil, fmt.Errorf("process discovery not supported on %s", runtime.GOOS)
}
